pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -319,7 +319,7 @@ func (c *Reconciler) resolvePipelineState(
 		if err != nil {
 			return nil, fmt.Errorf("failed to list VerificationPolicies from namespace %s with error %v", pr.Namespace, err)
 		}
-		fn := tresources.GetVerifiedTaskFunc(ctx, c.KubeClientSet, c.PipelineClientSet, c.resolutionRequester, pr, task.TaskRef, trName, pr.Namespace, pr.Spec.ServiceAccountName, vp)
+		fn := tresources.GetVerifiedTaskFunc(ctx, c.KubeClientSet, c.PipelineClientSet, c.resolutionRequester, pr, task.TaskRef, trName, pr.Namespace, pr.Spec.ServiceAccountName, vp, nil, &pr.Status)
```

Comment:
It looks like `GetVerifiedTaskFunc` updates the PipelineRun status. This isn't a function I'd expect to be mutating. Is there a better place for status updates?

Reply:
This is a pain point for me when I worked on this feature. The current PR's logic is to pass the status and update the status in `VerifyTask` `VerifyPipeline` `verifyResource`. 

If we want to set the status based on the return, then the tricky part may be when there's no error returned. 
It could be of the 4 cases we tested in `TestReconcile_verifyResolvedTask_Success`, and each case has different way of updating the status condition.  

So in the current implementation I passed the status and add the condition right after each case occurs. I thought this may have better readability?

Reply:
Thanks for the suggestion! I wonder if we could just return the condition from `GetVerifiedTaskFunc` and set the condition?

Reply:
I have changed to return a new type of error in the latest commit. Not sure if this is a correct approach cuz it affects lots of places where we check if the err is nil...